main.cpp

```diff
@@ -75,41 +112,38 @@ void dot_event()
             break;
         }
     }
-
 }
 
-
 /**
  * Connects to the Cellular Network
  */
 nsapi_error_t do_connect()
 {
-    nsapi_error_t retcode;
+    nsapi_error_t retcode = NSAPI_ERROR_OK;
     uint8_t retry_counter = 0;
 
     while (!iface.is_connected()) {
-
         retcode = iface.connect();
         if (retcode == NSAPI_ERROR_AUTH_FAILURE) {
             print_function("\n\nAuthentication Failure. Exiting application\n");
-            return retcode;
-        } else if (retcode != NSAPI_ERROR_OK) {
+        } else if (retcode == NSAPI_ERROR_OK) {
+            print_function("\n\nConnection Established.\n");
+        } else if (retry_counter > RETRY_COUNT) {
+            snprintf(print_text, PRINT_TEXT_LENGTH, "\n\nFatal connection failure: %d\n", retcode);
+            print_function(print_text);
+        } else {
             snprintf(print_text, PRINT_TEXT_LENGTH, "\n\nCouldn't connect: %d, will retry\n", retcode);
             print_function(print_text);
             retry_counter++;
             continue;
-        } else if (retcode != NSAPI_ERROR_OK && retry_counter > RETRY_COUNT) {
-            snprintf(print_text, PRINT_TEXT_LENGTH, "\n\nFatal connection failure: %d\n", retcode);
-            print_function(print_text);
-            return retcode;
         }
-
         break;
     }
+#if !MBED_CONF_MBED_TRACE_ENABLE
+    dot_thread.terminate();
```

Comment:
Do you really need to terminate? It's a bit hairy to forcibly terminate threads, particularly if they could be holding mutexes. If you need to shut it down, would be cleaner to send it an event telling it to kill itself "voluntarily", then join if necessary. Use `Thread::wait_event(4000)` to pick up that event rather than wait(4)

Reply:
Removed terminating of thread

Reply:
removed terminate